src/main/java/org/jenkinsci/plugins/urltrigger/content/JSONContentType.java

```diff
@@ -24,120 +25,124 @@
 
 	private transient Map<String, Object> results = null;
 
-    private List<JSONContentEntry> jsonPaths = new ArrayList<JSONContentEntry>();
+	private List<JSONContentEntry> jsonPaths = new ArrayList<JSONContentEntry>();
 
-    @DataBoundConstructor
-    public JSONContentType(List<JSONContentEntry> jsonPaths) {
-        if (jsonPaths != null) {
-            this.jsonPaths = jsonPaths;
-        }
-    }
+	@DataBoundConstructor
+	public JSONContentType(List<JSONContentEntry> jsonPaths) {
+		if (jsonPaths != null) {
+			this.jsonPaths = jsonPaths;
+		}
+	}
 
-    @SuppressWarnings("unused")
-    public List<JSONContentEntry> getJsonPaths() {
-        return jsonPaths;
-    }
+	@SuppressWarnings("unused")
+	public List<JSONContentEntry> getJsonPaths() {
+		return jsonPaths;
+	}
 
 	@Override
-    protected void initForContentType(String content, XTriggerLog log) throws XTriggerException {
+	protected void initForContentType(String content, XTriggerLog log) throws XTriggerException {
 		try {
 			JsonUtils.validateJson(content);
 			results = readJsonPath(content);
-		} catch( XTriggerException pe ) {
-			log.error( "An error occurred when parsing the document - may not be valid JSON?" ) ;
-			throw pe ;
+		} catch (XTriggerException pe) {
+			log.error("An error occurred when parsing the document - may not be valid JSON?");
+			throw pe;
+		}
+	}
+
+	private Map<String, Object> readJsonPath(String content) throws XTriggerException {
+		Map<String, Object> results = new HashMap<String, Object>(jsonPaths.size());
+		for (JSONContentEntry jsonContentEntry : jsonPaths) {
+			String jsonPath = jsonContentEntry.getJsonPath();
+			try {
+				Object result = JsonPath.read(content, jsonPath);
+				results.put(jsonPath, result);
+			} catch (PathNotFoundException pe) {
+				results.put(jsonPath, "");
```

Comment:
Rather than setting the value to an empty string, should we not make it null?  The logic already exists to handle this situation and it then specifically informs the user that the path was/is not present, rather than simply looking like a value change.

Reply:
Hi @TonyNoble,
I have introduced null instead of empty string now.

Best Regards
Kemal